src/extensions/mvs/camera.ts

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (c) 2023 mol* contributors, licensed under MIT, See LICENSE file for more info.
+ *
+ * @author Adam Midlik <midlik@gmail.com>
+ */
+
+import { Camera } from '../../mol-canvas3d/camera';
+import { GraphicsRenderObject } from '../../mol-gl/render-object';
+import { Sphere3D } from '../../mol-math/geometry';
+import { BoundaryHelper } from '../../mol-math/geometry/boundary-helper';
+import { Vec3 } from '../../mol-math/linear-algebra';
+import { Loci } from '../../mol-model/loci';
+import { Structure } from '../../mol-model/structure';
+import { PluginStateObject } from '../../mol-plugin-state/objects';
+import { PluginCommands } from '../../mol-plugin/commands';
+import { PluginContext } from '../../mol-plugin/context';
+import { StateObjectSelector } from '../../mol-state';
+import { ColorNames } from '../../mol-util/color/names';
+
+import { decodeColor } from './helpers/utils';
+import { ParamsOfKind } from './tree/generic/tree-schema';
+import { MolstarTree } from './tree/molstar/molstar-tree';
+import { MVSDefaults } from './tree/mvs/mvs-defaults';
+
+
+/** Defined in `../../mol-plugin-state/manager/camera.ts` but private */
+const DefaultCameraFocusOptions = {
+    minRadius: 5,
+    extraRadiusForFocus: 4,
+    extraRadiusForZoomAll: 0,
+};
+const DefaultCanvasBackgroundColor = ColorNames.white;
+
+
+/** Set the camera based on a camera node params. */
+export async function setCamera(plugin: PluginContext, params: ParamsOfKind<MolstarTree, 'camera'>) {
+    const target = Vec3.create(...params.target);
+    const position = Vec3.create(...params.position);
+    const up = Vec3.create(...params.up);
+    const snapshot: Partial<Camera.Snapshot> = { target, position, up };
+    await PluginCommands.Camera.SetSnapshot(plugin, { snapshot });
+}
+
+/** Focus the camera on the bounding sphere of a (sub)structure (or on the whole scene if `structureNodeSelector` is null).
+ * Orient the camera based on a focus node params. */
+export async function setFocus(plugin: PluginContext, structureNodeSelector: StateObjectSelector | undefined, params: ParamsOfKind<MolstarTree, 'focus'> = MVSDefaults.focus) {
+    let structure: Structure | undefined = undefined;
+    if (structureNodeSelector) {
+        const cell = plugin.state.data.cells.get(structureNodeSelector.ref);
+        structure = cell?.obj?.data;
+        if (!structure) console.warn('Focus: no structure');
+        if (!(structure instanceof Structure)) {
+            console.warn('Focus: cannot apply to a non-structure node');
+            structure = undefined;
+        }
+    }
+    const boundingSphere = structure ? Loci.getBoundingSphere(Structure.Loci(structure)) : getPluginBoundingSphere(plugin);
+    if (boundingSphere && plugin.canvas3d) {
```

Comment:
Would `this.plugin.managers.camera.focusSphere()` (or some other variant) work here?

And if not, possible extend that method with optional arguments to behave the same?

Reply:
Well, `this.plugin.managers.camera.focusSphere()` has a `principalAxes` option but I failed to understand what it actually does. I managed to pass some data to it, but it flips the orientation ad libitum. 
Plus, it cannot be awaited, which becomes a problem e.g. when you want to run it headless to render images. This could be solved by calling PluginCommands.Camera.Focus instead (and adding the option there), but it still doesn't solve the flipping.
To sum up, I think it's just easier to compute the snapshot and call `PluginCommands.Camera.SetSnapshot`.